File: single_number.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        
        unordered_map<int, int> freq_map;

        for (int n: nums){
            freq_map[n] ++;
        }  
        for (auto el: freq_map){
            if(el.second == 1){
                return el.first;
            }
        }
    return -1;
    }    
};

File: single_number.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:

        freq = {}

        for el in nums:
            freq[el] = 0
        for el in nums:
            freq[el] +=1

        for key in freq:
            if freq[key] == 1:
                return key

        return -1